code/modules/logging/log_holder.dm

```diff
@@ -67,4 +156,37 @@ GLOBAL_PROTECT(logger)
 	if(!log_category)
 		Log(LOG_CATEGORY_NOT_FOUND, message, data)
 		CRASH("Attempted to log to a category that doesn't exist! [category]")
-	log_category.add_entry(message, data)
+	log_category.add_entry(message, recursive_jsonify(data))
+
+/// Recursively converts an associative list of datums into their jsonified(list) form
+/datum/log_holder/proc/recursive_jsonify(list/data_list)
+	if(!data_list)
+		return null
+
+	var/list/jsonified_list = list()
+	for(var/key in data_list)
+		var/datum/data = data_list[key]
+		if(isnull(data))
+			stack_trace("recursive_jsonify called with a null value in the list")
+			continue
+
+		if(islist(data))
+			data = recursive_jsonify(data)
+
+		else if(isdatum(data))
+			var/list/options_list = list(
+				SCHEMA_VERSION = LOG_CATEGORY_SCHEMA_VERSION_NOT_SET,
+			)
+			var/list/serialization_data = data.serialize_list(options_list)
+			if(!length(serialization_data)) // serialize_list wasn't implemented, and errored
```

Comment:
Can we stack_trace here just in case we're dealing with a shitty `return null` or something else stupid?